lambda_functions/analyzer/binary_info.py

```diff
@@ -29,50 +30,71 @@ def __init__(self, bucket_name: str, object_key: str, yara_analyzer: YaraAnalyze
         self.download_path = os.path.join(
             tempfile.gettempdir(), 'binaryalert_{}'.format(uuid.uuid4()))
         self.yara_analyzer = yara_analyzer
-
+    
         # Computed after file download and analysis.
+        self.is_skipped = False
         self.download_time_ms = 0.0
         self.s3_last_modified = ''
         self.s3_metadata: Dict[str, str] = dict()
         self.computed_md5 = ''
         self.computed_sha = ''
         self.yara_matches: List[YaraMatch] = list()
+        
 
     def __str__(self) -> str:
         """Use the S3 identifier as the string representation of the binary."""
         return self.s3_identifier
 
     def _download_from_s3(self) -> None:
-        """Download binary from S3 and measure elapsed time."""
-        LOGGER.debug('Downloading %s to %s', self.object_key, self.download_path)
+        """Download binary from S3 and measure elapsed time, if it's under the size limit."""
+        file_size = analyzer_aws_lib.get_file_size_from_s3(self.bucket_name, self.object_key)
 
+        if file_size > self.MAX_FILE_SIZE_BYTES:
+            LOGGER.warning(f'File {self.object_key} is too large ({file_size} bytes). Skipping download.')
+            self.is_skipped = True
```

Comment:
Probably should return here instead of downloading the file even if it's going to be skipped.